webapp/src/components/admin_settings/teams_input/teams_input.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import debounce from 'lodash/debounce';
+import AsyncSelect from 'react-select/async';
+
+// TeamsInput searches and selects teams displayed by display_name.
+// Teams prop can handle the team object or strings directly if the team object is not available.
+// Returns the selected team ids in the `OnChange` value parameter.
+export default class TeamsInput extends React.PureComponent {
+    static propTypes = {
+        placeholder: PropTypes.string,
+        teams: PropTypes.array,
+        onChange: PropTypes.func,
+        actions: PropTypes.shape({
+            searchTeams: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    onChange = (value) => {
+        if (this.props.onChange) {
+            this.props.onChange(value);
+        }
+    }
+
+    getOptionValue = (team) => {
+        if (team.id) {
+            return team.id;
+        }
+
+        return team;
+    };
+
+    formatOptionLabel = (option) => {
+        if (option.display_name) {
+            return (
+                <React.Fragment>
+                    { `${option.display_name}`}
+                </React.Fragment>
+            );
+        }
+
+        return option;
+    }
+
+    searchTeams = debounce((term, callback) => {
+        this.props.actions.searchTeams(term).then(({data, error}) => {
+            if (error) {
+                // eslint-disable-next-line no-console
+                console.error('Error searching team in custom attribute settings dropdown.' + error.message);
```

Comment:
Non-blocking request to add a space after the first sentence
```suggestion
                console.error('Error searching team in custom attribute settings dropdown. ' + error.message);
```

Reply:
This is what happens when I don't use a code formatter :sweat_smile: